### DISCUSSION TITLE
Uplink in progress and LoRaMac State

### DISCUSSION THREAD

**Discussion by altishchenko** [Lora-net/LoRaMac-node#1172] [Q&A] [answered]
Hello Miguel!
(@mluis1)

I am sorry for the direct and narrow question.

Is it safe to assume that while the confirmed uplink is being processed by the stack and goes through the configured retransmissions, the LoRaMac MacState will reflect LORAMAC_STATUS_BUSY even if I attempt to perform a ReJoin at this time?

The reason I am asking is that, due to some legacy quirks of our software, we are locking the applications transmission queue until positive or negative confirmation is received via MCPS-Confirm event and I am worried that if I successfully trigger the ReJoin procedure this confirmation may be lost so the queue will stay in a locked condition forever. I tried a few experiments and **didn't** run into this possibility, but I would like to ask you to confirm:

> The LoRaMacMlmeRequest(), which is guarded by LoRaMacIsBusy(), will fail the ReJoin() request while a confirmed uplink is being processed and not yet *failed* due to the exceeding re-transmission limit. The stack will transition to Idle state *after* the appropriate MCPS-Confirm event will be issued to the application.

**Comment by mluis1** [comment, marked as answer]
@altishchenko Yes, you can assume that no new transmission requests will be processed by the stack up until the previous request has been fully processed.

If one observes a different behavior it would mean that we have introduced an involuntary issue.
